clang/lib/Sema/SemaExpr.cpp

```diff
@@ -16597,7 +16597,8 @@ ExprResult Sema::ActOnBlockStmtExpr(SourceLocation CaretLoc,
   BD->setCaptures(Context, Captures, BSI->CXXThisCaptureIndex != 0);
 
   // Pop the block scope now but keep it alive to the end of this function.
-  AnalysisBasedWarnings::Policy WP = AnalysisWarnings.getDefaultPolicy();
+  AnalysisBasedWarnings::Policy WP =
+      AnalysisWarnings.getPolicyInEffectAt(Body->getEndLoc());
```

Comment:
Static analysis points out that `Body` is nullptr checked above and so we should not be unconditionally dereferencing it here.

Reply:
I missed the `cast`, yeah not bug then.